extensions/db-diagram/src/models/ITableModel.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export default interface ITableModel {
+
+	//Name of the table
+	name: string;
+
+	//Summary of the table
+	summary: string;
```

Comment:
If by summary you mean the text that you plan to show in the "contextual view", then you should probably get rid of this here. The summary should be generated on the extension side, and so you should be able to generate the summary from the results you get from sqltoolsservice.
If we are using the insights generator or any other library to create the summaries, then look into what inputs for those libraries are, and fetch that information instead. For now, if you just want to show the relationships, the relationships field might suffice.

Reply:
Yeah, I meant the summary field to act as the text that would populate the "summary" card within the "contextual" view. For right now, to have placeholder data, the summary input was going to be passed in the constructor. However, I agree, later on, the summary should be generated from either the results from sqltoolsservice or from the insights team. 
Given the uncertainty there, I was planning to create an "initializeSummary" helper method within the tableModel class that would be called in the tableModel constructor. The intention for the method is to either take in inputs from sqltoolsservice and generate a summary, or use the insights library to generate a summary.

Reply:
Right now I'm thinking mainController.ts should
- register the DBDiag command
- call sqltoolsservice and/or insights generator to initialize data models
- render UI using modelViewController with input data taken from data models

Reply:
class mainController {

register DB Diag Command
call DiagramService from sqltoolsservice, then instantiate a tableModel object and a dbModel object
render UI

}

Reply:
Yes, that seems like the right idea. You can finalize the UI and then paste a screenshot here while I work on the backend to tie your models up with your UI.